app/controllers/api/functions.php

```diff
@@ -1376,6 +1389,9 @@
             $queries[] = Query::search('search', $search);
         }
 
+        // Apply internal queries
+        $queries[] = Query::equal('functionInternalId', [$function->getInternalId()]);
```

Comment:
What exactly does this line do?

Reply:
If it was not there, you would get **all** variables. I have a few functions, this is what my "variables" tab look like: (same on all of them)
![CleanShot 2022-09-11 at 13 57 47](https://user-images.githubusercontent.com/19310830/189526279-7c588fc1-b679-4f07-8e57-dcba88a07495.png)

Reply:
Got it. Thanks

Reply:
Wait, why aren't we using a subquery to get all the vars?!

Reply:
To support pagination. As part of feature of 1.0 - queries on all list endpoints. Stuff like

```
Query.equal("name", "API_KEY")
```
(when you want to find by name, instead of by ID)

Reply:
Why don't we do it for sessions?

Reply:
I doubt we really need pagination for this kind of entity. With the new console we didn't even consider it.

Reply:
Understandable. Let's remove queries support from this endpoint and use subQueries.

Reply:
Removed queries and search feature from variables, and used subquery in the listVariables endpoint.